client/src/main/scala/skuber/api/client/token/AwsAuthRefreshable.scala

```diff
@@ -0,0 +1,97 @@
+package skuber.api.client.token
+
+import java.net.URI
+import java.util.{Base64, Date}
+import com.amazonaws.DefaultRequest
+import com.amazonaws.auth._
+import com.amazonaws.auth.presign.{PresignerFacade, PresignerParams}
+import com.amazonaws.http.HttpMethodName
+import com.amazonaws.internal.auth.DefaultSignerProvider
+import com.amazonaws.regions.Regions
+import com.amazonaws.services.securitytoken.model.GetCallerIdentityRequest
+import com.amazonaws.services.securitytoken.{AWSSecurityTokenServiceClient, AWSSecurityTokenServiceClientBuilder}
+import org.joda.time.DateTime
+import skuber.K8SException
+import skuber.api.client._
+import scala.concurrent.duration._
+
+final case class AwsAuthRefreshable(cluster: Option[Cluster] = None) extends AuthProviderRefreshableAuth {
+
+  @volatile var cachedToken: Option[RefreshableToken] = None
+
+  private val clusterName: String = {
+    cluster.flatMap(_.clusterName).getOrElse {
+      defaultK8sConfig.currentContext.cluster.clusterName.getOrElse {
+        throw new K8SException(Status(reason =
+          Some("Cluster name not found, please provide an EKS (AWS) cluster name within Cluster object")))
+      }
+    }
+  }
+
+  //  https://github.com/kubernetes-sigs/aws-iam-authenticator/blob/27337b2b74c3140cf745a64f7154fe8ff7592258/pkg/token/token.go#L87
+  // STS provides 15 minutes expiration, and it's not configurable.
+  // Using 10 minutes to be on the safe side.
+  private val refreshInterval = 10.minutes
+
+  override def refreshToken: RefreshableToken = {
+    val token = generateToken
+    val refreshedToken = RefreshableToken(token, DateTime.now.plus(refreshInterval.toMillis))
+    cachedToken = Some(refreshedToken)
+    refreshedToken
+  }
+
+  override def generateToken: String = {
+    try {
+      val credentialsProvider = new AWSStaticCredentialsProvider(new DefaultAWSCredentialsProviderChain().getCredentials)
+      val signer: Signer = SignerFactory.createSigner(SignerFactory.VERSION_FOUR_SIGNER,
+        new SignerParams("sts", Regions.US_EAST_1.getName)) //it needs to be "us_east_1"
+
+      val tokenService: AWSSecurityTokenServiceClient = AWSSecurityTokenServiceClientBuilder
+        .standard()
+        .withCredentials(credentialsProvider)
+        .build().asInstanceOf[AWSSecurityTokenServiceClient]
+
+      val callerIdentityRequestDefaultRequest = new DefaultRequest[GetCallerIdentityRequest](new GetCallerIdentityRequest(), "sts")
+      val uri = new URI("https", "sts.amazonaws.com", null, null)
```

Comment:
Per AWS calls too STS should be regionalized. ref https://docs.aws.amazon.com/sdkref/latest/guide/feature-sts-regionalized-endpoints.html. in this case $"sts.$region.amazonaws.com".

Reply:
that's true, but it still works with sts.amazonaws.com
with this approach we don't need to require from the users the region name and also there are few regions that don't have sts deployed.
https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_temp_enable-regions.html
eu-south-1
me-south-1
ap-southeast-3
ap-east-1
af-south-1

Reply:
We should allow for the region to be specified and if it is not specified that default to the global. We can't rely on the global for our needs.

Reply:
agreed i will fix it